apis/v1alpha2/nginxproxy_types.go

```diff
@@ -412,6 +418,47 @@ type DaemonSetSpec struct {
 	Container ContainerSpec `json:"container"`
 }
 
+type HPASpec struct {
```

Comment:
Sorry - this comment went missing in my last review! This enables validation to ensure we have at least one metrics source, requires the percentages are between 1 and 100, and ensures min replicas is less than max replicas
```suggestion
// +kubebuilder:validation:XValidation:message="at least one metric must be specified when autoscaling is enabled",rule="!self.enabled || (has(self.targetCPUUtilizationPercentage) || has(self.targetMemoryUtilizationPercentage) || (has(self.autoscalingTemplate) && size(self.autoscalingTemplate) > 0))"
// +kubebuilder:validation:XValidation:message="minReplicas must be less than or equal to maxReplicas",rule="self.minReplicas <= self.maxReplicas"
// +kubebuilder:validation:XValidation:message="CPU utilization must be between 1 and 100",rule="!has(self.targetCPUUtilizationPercentage) || (self.targetCPUUtilizationPercentage >= 1 && self.targetCPUUtilizationPercentage <= 100)"
// +kubebuilder:validation:XValidation:message="memory utilization must be between 1 and 100",rule="!has(self.targetMemoryUtilizationPercentage) || (self.targetMemoryUtilizationPercentage >= 1 && self.targetMemoryUtilizationPercentage <= 100)"
type HPASpec struct {
```